pkg/gitops/filter.go

```diff
@@ -0,0 +1,80 @@
+package gitops
+
+import (
+    "fmt"
+	"time"
+    "strings"
+    "path/filepath"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+func (cr *CustomRepo) FilterCommits(author *string, since *time.Time, until *time.Time, resource *string, namespace *string, name *string) ([]object.Commit, error) {
```

Comment:
I think we should use the default values to mean "do not filer" and we don't need pointers at all here